docs/DeveloperGuide.md

```diff
@@ -262,73 +262,227 @@ _{Explain here how the data archiving feature will be implemented}_
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+**MSS**
 
-*{More to be added}*
+1. User requests to list projects
+2. HRConnect shows a list of projects with their ids
+3. User requests to delete a specific project in the list by its id
+4. HRConnect deletes the project
 
-### Glossary
+   Use case ends.
 
-* **Mainstream OS**: Windows, Linux, Unix, MacOS
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+**Extensions**
 
---------------------------------------------------------------------------------------------------------------------
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 3a. The given id is invalid.
+
+  - 3a1. HRConnect shows an error message.
+
+    Use case resumes at step 2.
+
+---
+
+**Use case: Assign an employee to a project**
+
+**MSS**
+
+1. User requests to list employees
+2. HRConnect shows a list of employees with their ids
+3. User requests to list projects
+4. HRConnect shows a list of projects with their ids
+5. User requests to assign a specific employee by their id to a specific project by its id
+6. HRConnect assigns the employee to the project
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 4a. The list is empty.
+
+  Use case ends.
+
+- 5a. Either of the given ids is invalid.
+
+  - 5a1. HRConnect shows an error message.
+
+    HRConnect displays the list of projects if the project id is invalid and the list of employees if the employee id is invalid. Use case resume at step 4.
+
+---
+
+**Use case: Un-assign an employee from a project**
+
+**MSS**
+
+1. User requests to list projects
+2. HRConnect shows a list of projects with their ids
+3. User selects a project by its id
+4. HRConnect shows a list of employees assigned to the project
+5. User requests to un-assign a specific employee by their id from the selected project
+6. HRConnect un-assigns the employee from the project
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list is empty.
+
+  Use case ends.
+
+- 4a. The list is empty.
+
+  Use case ends.
+
+- 5a. The given id is invalid.
+
+  - 5a1. HRConnect shows an error message.
+
+    Use case resumes at step 4.
+
+---
+
+**Use case: Filter employees with desired skills**
+
+**MSS**
+
+1. User requests to list skills the employees in the company have
+2. HRConnect shows a list of skills
+3. User selects a few skills
+4. User requests to list all employees with all of the skills selected
+5. HRConnect lists all employees with all of the skills selected
+
+   Use case ends.
+
+**Extensions**
+
+- 2a. The list of skills is empty.
+
+  Use case ends.
+
+- 3a. User selects invalid skills.
+
+  - 3a1. HRConnect shows an error message.
+
+    Use case resumes at step 2.
```

Comment:
Use cases seem good! They seem to follow requirements in textbook well.
I would use periods at the end of each line for formatting consistency, but overall LGTM!

Reply:
Thanks. Updated the document to add periods at the end of each line.

Reply:
Thanks! Looks great now